File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
	def initialize(num)
		@num = num
	end

	def square_of_sum
		num = @num

		square = (1..num).reduce(0) {|total, num|
			total += num
			total
		}**2
	end

	def sum_of_squares
		num = @num

		square = (1..num).reduce(0) {|total, num|
			total += num**2
			total
		}
	end

	def difference
		square_of_sum - sum_of_squares
	end

end

  module BookKeeping
    VERSION = 3 # Where the version number matches the one in the test.
  end

File: javascript/difference-of-squares/difference-of-squares.js
var Squares = function(n){
  this.squareOfSums = this.squareOfSums(n);
  this.sumOfSquares = this.sumOfSquares(n);
  this.difference = this.squareOfSums - this.sumOfSquares;
}

Squares.prototype.squareOfSums = function(n){
  var sum = 0;

  for(var i = 0; i <= n; i++){
    sum += i;
  }

  return Math.pow(sum, 2);
}

Squares.prototype.sumOfSquares = function(n){
  var sum = 0;

  for(var i = 0; i <= n; i++){
    sum += Math.pow(i, 2);
  }

  return sum;
}

module.exports = Squares;
